plugins/outputs/cratedb/cratedb.go

```diff
@@ -68,15 +71,31 @@ CREATE TABLE IF NOT EXISTS ` + c.Table + ` (
 func (c *CrateDB) Write(metrics []telegraf.Metric) error {
 	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(c.Timeout))
 	defer cancel()
-	if sql, err := insertSQL(c.Table, metrics); err != nil {
+
+	generatedSQL, err := insertSQL(c.Table, c.KeySeparator, metrics)
+	if err != nil {
 		return err
-	} else if _, err := c.DB.ExecContext(ctx, sql); err != nil {
+	}
+
+	result, err := c.DB.ExecContext(ctx, generatedSQL)
+	if err != nil {
 		return err
 	}
+
+	if affected, err := result.RowsAffected(); err != nil {
+		return err
+	} else if affected != int64(len(metrics)) {
+		// Unfortunately this is the most we can do to ensure everything was written. CrateDB
+		// does not return an error for a single row when doing bulk inserts, instead it just
+		// decrements the number of affected rows.
+		// https://crate.io/docs/crate/reference/en/latest/general/dml.html#inserting-data
+		return fmt.Errorf("failed to insert %d rows", int64(len(metrics))-affected)
```

Comment:
only return an error if you want these writes to be retried until they succeed. If there are valid reasons for some rows not to be written to the db (eg they already exist there) then this will fail unnecessarily.

Reply:
What do you suggest we do then, just keep it as it was or something else?

Reply:
probably shouldn't check RowsAffected; error isn't necessary